From: Gunnar Wrobel <p@rdus.de>
Subject: [PATCH] t/turba/HK/GW/AllowAttributeDefaults

Allow attribute default values for the turba source configuration.

kolab/issue3439 (FreeBusy display in web client and fbview depends on kolabHomeServer in LDAP)

ISSUE: https://www.intevation.de/roundup/kolab/issue3439

STATUS: SUBMIT

Signed-off-by: Gunnar Wrobel <p@rdus.de>

---
 horde-webmail/turba/config/sources.php |    3 ++-
 horde-webmail/turba/lib/Object.php     |    6 +++++-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/horde-webmail/turba/config/sources.php b/horde-webmail/turba/config/sources.php
index 6eb2adc..5e4df9d 100644
--- a/horde-webmail/turba/config/sources.php
+++ b/horde-webmail/turba/config/sources.php
@@ -332,7 +332,8 @@ if (!empty($GLOBALS['conf']['kolab']['enabled'])) {
                 'notes'             => 'description',
                 'kolabHomeServer'   => 'kolabHomeServer',
                 'freebusyUrl'       => array('fields' => array('kolabHomeServer', 'email'),
-                                             'format' => 'https://%s/freebusy/%s.xfb'),
+                                             'format' => 'https://%s/freebusy/%s.xfb',
+                                             'defaults' => array('kolabHomeServer' => $GLOBALS['conf']['kolab']['imap']['server'])),
             ),
             'search' => array(
                 'name',
diff --git a/horde-webmail/turba/lib/Object.php b/horde-webmail/turba/lib/Object.php
index 57c9368..68b537d 100644
--- a/horde-webmail/turba/lib/Object.php
+++ b/horde-webmail/turba/lib/Object.php
@@ -115,7 +115,11 @@ class Turba_Object {
             is_array($this->driver->map[$attribute])) {
             $args = array();
             foreach ($this->driver->map[$attribute]['fields'] as $field) {
-                $args[] = $this->getValue($field);
+                $value = $this->getValue($field);
+                if (empty($value) && isset($this->driver->map[$attribute]['defaults'][$field])) {
+                    $value = $this->driver->map[$attribute]['defaults'][$field];
+                }
+                $args[] = $value;
             }
             return trim(vsprintf($this->driver->map[$attribute]['format'], $args), " \t\n\r\0\x0B,");
         } elseif (!isset($this->attributes[$attribute])) {
-- 
tg: (2e6e530..) t/turba/HK/GW/AllowAttributeDefaults (depends on: t/Kolab_Storage/HK/GW/TestingFixes)
-- 
TOPGIT patch commit log
=======================

commit d3e51ed7f0eb792de6a84975e6ae51199ed1b7cb
Author: Gunnar Wrobel <p@rdus.de>
Date:   Sun Apr 26 12:55:30 2009 +0200

    Allow attribute default values for the turba source configuration.
    
    kolab/issue3439 (FreeBusy display in web client and fbview depends on kolabHomeServer in LDAP)
    
    ISSUE: https://www.intevation.de/roundup/kolab/issue3439
